src/NuGet.Clients/PackageManagement.VisualStudio/ProjectSystems/ProjectSystemCache.cs

```diff
@@ -130,6 +148,8 @@ public void RemoveProject(string name)
             {
                 _readerWriterLock.ExitUpgradeableReadLock();
             }
+
+            FireCacheUpdatedEvent();
```

Comment:
We should probably fire the event only in `AddProjectRestoreInfo` which is called via the nomination. This'd be just enough to resolve the bug. Not sure what side effects will be with so many events raised.